metaflow/plugins/argo/argo_live_run.py

```diff
@@ -0,0 +1,248 @@
+from metaflow.plugins.argo.argo_client import ArgoClient
+from metaflow.metaflow_config import KUBERNETES_NAMESPACE
+import time
+
+
+class ArgoLiveRun:  # TODO: make child class of LiveRun
+    """
+    This class takes in information to identify and trigger a run of a
+    Metaflow flow using the Argo plugin. The run is initialized without
+    triggering a run, but a run should be triggered immediately after
+    initialization. This class should be initialized only with the trigger_run
+    function. The flow to trigger is identified based on the parameter for the
+    Metaflow flow name. Users may use additional parameters to alter the run,
+    and wait for the run to finish.
+
+    The object allows users to access information relating to the triggered
+    run, including booleans for whether the run has been triggered, is running,
+    and was successful, as well as for failed steps, and exceptions.
+    """
+
+    def __init__(
+        self,
+        flow_name: str = None,
+        alt_flow_id_info: dict = None,
+        parameters: dict = None,
+        wait: bool = True,
+        wait_timeout: int = 30,  # in minutes TODO: determine proper timeout
+    ):
+        """
+        Initialize a TriggeredRun object without triggering flow.
+
+        Parameters
+        ----------
+        flow_name: str
+            Name of Metaflow flow to trigger run
+        alt_flow_id_info: dict
+            Alternative identifying information of flow to trigger run
+        parameters: dict
+            The information passed in to affect how run is triggered
+        wait: bool
+            whether function waits for triggered run to finish before returning
+        wait_timeout: int
+            time (in minutes) to wait for run to finish
+        """
+        if alt_flow_id_info is None:
+            alt_flow_id_info = {}
+
+        if parameters is None:
+            parameters = {}
+
+        # initialize instance variables
+        self._flow_name = flow_name
+        self._alt_flow_id_info = alt_flow_id_info
+        self._template_name = self._flow_name.lower()  # TODO: add logic when adding alt flow IDs
+        self._parameters = parameters
+        self._wait = wait
+        self._wait_timeout = wait_timeout
+        self._argo_client = ArgoClient(KUBERNETES_NAMESPACE)
+        self._flow_information = None
+        self._exception = None
+        self._cached_status = None
+        self._has_triggered = False
+        self._metaflow_run = None
+        self._argo_run_id = None
+        self._metaflow_run_id = None
+        self._kubernetes_namespace = None
+        self._failed_steps = None
+
+    def trigger(self) -> None:
```

Comment:
To make sure only 1 run is created: @classmethod, take all parameters for __init__, and return the newly created `ArgoLiveRun`.

Reply:
ArgoLiveRun.trigger() instead of ArgoLiveRun() to make the "trigger" behavior explicit.